src/poliastro/plotting.py

```diff
@@ -198,6 +199,55 @@ def plot(self, orbit, label=None, color=None):
 
         return lines
 
+    def set_lines_up_to_instant(frame, self, x, y, lines):
+        """Plots a frame of the precomputed trajectory.
+
+        Parameters
+        ----------
+        frame : current plotting frame.
+        x : x values of the trajectory to plot.
+        y : y values of the trajectory to plot.
+        lines : plot information
+        """
+        lines.set_data(x[:int(frame)], y[:int(frame)])
+        return lines
+
+    def animate(self, orbit, label=None, color=None):
+        """Plots state and osculating animated orbit in their plane.
+
+        """
+        if not self._frame:
+            self.set_frame(*orbit.pqw())
+
+        if (orbit, label) not in self._orbits:
+            self._orbits.append((orbit, label))
+
+        self.set_attractor(orbit.attractor)
+        self._redraw_attractor(orbit.r_p * 0.15)  # Arbitrary Threshhold
+
+        _, positions_in_tuples = orbit.sample(self.num_points)
+        positions = positions_in_tuples.get_xyz().transpose()
+        x, y = self._project(positions)
+
+        # The following should be in a init_animation function, but I [aeronau] still don't know how to pass the arguments to the init function
+        xmax = np.amax(x.to(u.km).value)
+        xmin = np.amin(x.to(u.km).value)
+        ymax = np.amax(y.to(u.km).value)
+        ymin = np.amin(y.to(u.km).value)
+        self.ax.set_xlim(xmin, xmax)
+        self.ax.set_ylim(ymin, ymax)
+
+        lines = []
+        a, = self.ax.plot([], [], '--', color=color, label=label)
+
+        ani = FuncAnimation(self.fig, self.set_lines_up_to_instant, frames=np.linspace(0, len(x) - 1, len(x)), fargs=(self, x.to(u.km).value, y.to(u.km).value, a), blit=True, save_count=len(x))
+
+        lines.append(a)
+
+        self.ax.set_xlabel("$x$ (km)")
+        self.ax.set_ylabel("$y$ (km)")
+        self.ax.set_aspect(1)
+        return lines
```

Comment:
Why not returning the `ani` instead?

And also, since matplotlib 2.1.0 we can display animations in notebooks! https://matplotlib.org/users/prev_whats_new/whats_new_2.1.0.html#interactive-js-widgets-for-animation

Reply:
In the future, maybe it makes sense to have a different object, that inherits some common functionality from OrbitPlotter. But this requires coordination with #338 and @shreyasbapat, so let's keep it simple for now.